src/systemd-sonic-generator/systemd-sonic-generator.c

```diff
@@ -0,0 +1,301 @@
+#include <assert.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <errno.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <linux/limits.h>
+
+#define MAX_NUM_TARGETS 5
+#define MAX_NUM_INSTALL_LINES 5
+#define MAX_NUM_UNITS 128
```

Comment:
How have these constants been determined?

Reply:
Just arbitrarily, they're just upper bounds, I don't think we have any services that actually go up to the limits.

Reply:
OK. I'm just concerned we might hit MAX_NUM_TARGETS or MAX_NUM_INSTALL_LINES in the future, as they are small, but I guess we'll catch it since you're printing pretty clear error messages.